public/scripts/controllers/events.js

```diff
@@ -12,6 +13,33 @@ __get_all_events = function ($http, on_success) {
     }
 }
 
+{
+    let get_current_event = () => {
+        var xhttp = new XMLHttpRequest();
+        xhttp.onreadystatechange = () => {
+            if (xhttp.readyState === 4) {
+                if (xhttp.status === 200) {
+                    currentEvent = JSON.parse(xhttp.responseText).event;
+                    currentEvent.addinfo_json = JSON.parse(currentEvent.addinfo_json);
+
+                } else {
+                    alert(`getting current event details failed`);
+                    console.error(`${response.status} - ${response.responseText}`);
+                }
+            }
+        };
+
+        let _url = '/events/current';
+        xhttp.open("GET", _url, false);
+        xhttp.send();
+        // keep as non sync, since we need to query the current event details before we use it.
+    }
+
+    currentEvent = async function () {
+        await get_current_event();
+    }
```

Comment:
currentEvent enable easy getting of current event details based on the user current event id (from session).
currentEvent will start as a function reference (to prevent unnecessary calls).
once currentEvent() is called, currentEvent will change into an object reference containing the current event details.

```
if (typeof currentEvent === 'function') { //save check to confirm mutation - will only happen once
    currentEvent();
}
currentEvent.propertyName    //as object
```